src/borg/archiver.py

```diff
@@ -1752,11 +1748,8 @@ def do_debug_dump_manifest(self, args, repository, manifest, key):
 
         meta = prepare_dump_dict(msgpack.fallback.unpackb(data, object_hook=StableDict, unicode_errors='surrogateescape'))
 
-        if args.path == '-':
-            json.dump(meta, sys.stdout, indent=4)
-        else:
-            with open(args.path, 'w') as fd:
-                json.dump(meta, fd, indent=4)
+        with dash_open(args.path, 'w') as fd:
+            json.dump(meta, fd, indent=4)
```

Comment:
instead of just offering dash_open function, we could have a full context manager that closes real files, but lets the streams open on exit.